File: Solution/4.median-of-two-sorted-arrays.go
package LeetCode

import "sort"

/*
 * @lc app=leetcode id=4 lang=golang
 *
 * [4] Median of Two Sorted Arrays
 */

// @lc code=start
func findMedianSortedArrays(nums1 []int, nums2 []int) float64 {
	var (
		// Combine both arrays into one array
		num []int = append(nums1, nums2...)
		// Find the middle index of the combined array
		m int = len(num) / 2
	)

	// Sort the combined array in ascending order
	sort.Ints(num)

	// Check if the combined array has odd or even number of elements
	if len(num)%2 == 1 {
		// If the combined array has odd number of elements, return the middle
		// element
		return float64(num[m])
	}

	// If the combined array has even number of elements, return the average of
	// the two middle elements
	return float64(num[m-1]+num[m]) / 2
}

// @lc code=end

File: 4.median-of-two-sorted-arrays.py
#
# @lc app=leetcode id=4 lang=python
#
# [4] Median of Two Sorted Arrays
#


# @lc code=start
class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: float
        """
        self.num = sorted(nums1 + nums2)

        self.m = len(self.num)

        if self.m % 2 == 0:
            return (float(self.num[self.m / 2 - 1]) + float(self.num[self.m / 2])) / 2

        return self.num[self.m / 2]


# @lc code=end
